keps/sig-auth/4940-psa-block-host-field-in-probes/README.md

```diff
@@ -0,0 +1,370 @@
+https://storage.googleapis.com/k8s-triage/index.html?test=TestPodSecurity
+
+The integration tests will verify the PSA policy validation logic by:
+- Creating test cases for each probe type (HTTPGet, TCPSocket) in a pod
+- Testing each probe location (LivenessProbe, ReadinessProbe, StartupProbe, LifecycleHandler)
+- Verifying the PSA policy enforcement at the baseline level
+- Testing the behavior with different PSA configurations
+
+##### e2e tests
+
+There are no Pod Security specific E2E tests (we rely on integration test coverage instead), but the Pod Security admission controller is enabled in E2E clusters, and all E2E test namespaces are labeled with the enforcement label for Pod Security.
+
+### Graduation Criteria
+
+The PSA added will be done within a single release
+and given there will be no feature gates for that,
+there is no need for multi-release graduation criteria.
+All related code will land within the same single release
+
+
+#### Deprecation
+
+`.Host` field will be deprecated from the Probe and Lifecycle Handlers
+
+### Upgrade / Downgrade Strategy
+
+Any older pods with this field set should not be affected
+with the above solution. Only newer pods getting created
+with the field will be alerted.
+
+Users who are using this field can switch to using exec
+probes moving forward which should unblock them given exec
+probes can provide the same functionality.
+
+
+### Version Skew Strategy
+
+N/A since its only within a single component: pod-security-admission
+and doesn't cross multiple components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+We decided to not go with feature gates and use PSA versioning.
+So if the admin sets pod-security.kubernetes.io/enforce-version: v1.34
+on a namespace this feature will get enabled.
+
+###### Does enabling the feature change any default behavior?
+
+* There is no effect on clusters where PSA is not enabled OR an older
+PSA version is used.
+
+* There is no effect on clusters where `.Host` probes are not used
+
+* There is no effect on clusters where an older PSA versioning is being
+used
+
+* If users create pod probes with `.Host` field set and the admin
+has set baseline PSA level to `enforce` mode then the request will be
+actively blocked and rejected. Existing pods with `.Host` probes
+that are upgrading will not be impacted.
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+No
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+N/A
+
+###### Are there any tests for feature enablement/disablement?
+
+N/A since there is no feature gate
+
+### Rollout, Upgrade and Rollback Planning
+
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+N/A
+
+###### What specific metrics should inform a rollback?
+
+N/A
```

Comment:
Find the PSA denials metrics to help cluster-admins determine if they suddenly got a problem.

Reply:
ah you meant https://kubernetes.io/docs/concepts/security/pod-security-admission/#metrics ?
thanks for mentioning this.. learnt something new, didn't know these existed.
added them